mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -412,163 +402,183 @@ struct BlockwiseGemmV2RewritePattern
+      if (KPack > 1) {
+        sourceOffsetA = ilkb.create<MulIOp>(
+            loc, sourceOffsetA, ilkb.create<ConstantIndexOp>(loc, KPack));
+        sourceOffsetB = ilkb.create<MulIOp>(
+            loc, sourceOffsetB, ilkb.create<ConstantIndexOp>(loc, KPack));
+        Value valueA = ilkb.create<InBoundsLoadOp>(
+            loc, bufferAElementType, op.getMatrixA(), sourceOffsetA);
+        ilkb.create<memref::StoreOp>(loc, valueA, bufferA,
+                                     ValueRange{zeroConstantOp});
+        Value valueB = ilkb.create<InBoundsLoadOp>(
+            loc, bufferBElementType, op.getMatrixB(), sourceOffsetB);
+        ilkb.create<memref::StoreOp>(loc, valueB, bufferB,
+                                     ValueRange{zeroConstantOp});
 
-      Value valueB = ilnkb.create<InBoundsLoadOp>(
-          loc, bufferBElementType, op.getMatrixB(), sourceOffsetB);
-      ilnkb.create<memref::StoreOp>(loc, valueB, bufferB,
-                                    ValueRange{destOffsetB});
+      } else {
+        auto innerLoopKBase = ilkb.create<AffineForOp>(loc, 0, k_base);
+        auto ilkbase = ConversionPatternRewriter::atBlockBegin(
+            innerLoopKBase.getBody(), olmb.getListener());
+        auto ilkbaseiv = innerLoopKBase.getInductionVar();
+
+        sourceOffsetA = ilkbase.create<AddIOp>(
+            loc, sourceOffsetA,
+            ilkbase.create<MulIOp>(loc, ilkbaseiv, MConstantOp));
+        sourceOffsetB = ilkbase.create<AddIOp>(
+            loc, sourceOffsetB,
+            ilkbase.create<MulIOp>(loc, ilkbaseiv, NConstantOp));
+        Value valueA = ilkbase.create<InBoundsLoadOp>(
+            loc, bufferAElementType, op.getMatrixA(), sourceOffsetA);
+        ilkbase.create<memref::StoreOp>(loc, valueA, bufferA,
+                                        ValueRange{zeroConstantOp});
+        Value valueB = ilkbase.create<InBoundsLoadOp>(
+            loc, bufferBElementType, op.getMatrixB(), sourceOffsetB);
+        ilkbase.create<memref::StoreOp>(loc, valueB, bufferB,
+                                        ValueRange{zeroConstantOp});
+      }
     } else {
       // const index_t blk_id = laneId / mfma_type.num_threads_blk;
       // const index_t blk_td = laneId % mfma_type.num_threads_blk;
       auto inputSpanLenConstantOp =
-          b.create<ConstantIndexOp>(loc, inputSpanLen);
-      auto blk_id = b.create<DivUIOp>(loc, laneId, inputSpanLenConstantOp);
-      auto blk_td = b.create<RemUIOp>(loc, laneId, inputSpanLenConstantOp);
+          ilkb.create<ConstantIndexOp>(loc, inputSpanLen);
+      auto blk_id = ilkb.create<DivUIOp>(loc, laneId, inputSpanLenConstantOp);
+      auto blk_td = ilkb.create<RemUIOp>(loc, laneId, inputSpanLenConstantOp);
 
-      Value kBaseA = b.create<AddIOp>(loc, aBase, blk_td);
-      Value kBaseB = b.create<AddIOp>(loc, bBase, blk_td);
+      Value kBaseA = ilkb.create<AddIOp>(loc, aBase, blk_td);
+      Value kBaseB = ilkb.create<AddIOp>(loc, bBase, blk_td);
```

Comment:
This should be `mOffset` instead of `aBase` and `nOffset` instead of `bBase`. If you make this change, this'll work also for the reduction case.

Reply:
Agreed, once your PR merges, we can add this additional offset from `mOffset` to ensure the added case will be enabled.

I will point out that as the code base in `develop` branch today, leaving this as should not break any pre-existing config (because we wouldn't select reduction case with multiple repeats).